### PR TITLE
Remove default rows in the experiment stepper

### PR DESCRIPTION
**Pull request by zackcl**
Changes:
  * Remove default rows in all tables in the stepper
  * Handle if the `conditionCode` is null, and if the user adds spaces to the condition when checking the `defaultKeyword`
  * Unfilled rows in the Participants step will be removed when the user clicks the "Next" or "Update" button (temporary solution)

Note: I will later add validation messages to the Participants step so we can show a message when the user clicks the "Next" button while `ID/Name` field is empty. Just like how it works in the Metrics step.

Screen Shot 2022-08-08 at 11 53 42 AM
Screen Shot 2022-08-08 at 11 55 33 AM

### CODE REVIEW

**Comment by zackcl** [comment]
@ppratikcr7 Please check if I properly removed the default rows. Thanks!

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts [hunk 1]
```diff
@@ -355,7 +353,7 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
     let defaultConditionCodeErrorText = this.translate.instant('home.new-experiment.design.condition-name-validation.text');
     if (conditions.length) {
       const hasDefaultConditionCode = conditions.filter(
-        condition => condition.conditionCode.toUpperCase() === defaultKeyword
+        condition => condition.conditionCode && condition.conditionCode.toUpperCase().trim() === defaultKeyword
```
[the remark this answers is not in the document]
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:356]: unless this for sure would only ever be a string

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts [hunk 2]
```diff
@@ -205,10 +202,12 @@ export class ExperimentParticipantsComponent implements OnInit {
 
     if (includedMembersFiltered.length === 0) {
       this.members1.clear();
+      this.updateView1();
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts:205]: what happens if this isn't included?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts [hunk 3]
```diff
@@ -101,8 +101,6 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
       this.isContextChanged = false;
       this.partition.clear();
       this.condition.clear();
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: how come we remove these? what did they do exactly?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts [hunk 4]
```diff
@@ -519,7 +519,6 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
     if (this.isContextChanged) {
       this.isContextChanged = false;
       this.queries.clear();
-      this.queries.push(this.addMetric());
```
Comment (danoswaltCL) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts:522]: same as other Q, what does removing this do?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:356]: I think your suggestion is more accurate, I will change that.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts (hunk 3 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: What do you mean? I didn't remove the 3 lines you mentioned. 
```
this.isContextChanged = false;
this.partition.clear();
this.condition.clear();
```

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts (hunk 2 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts:205]: So the problem with the previous code without the `updateView1()` is that if we create an empty row and move to the next step, it clears the `members1` table but it is not applied in the UI so even if you fill the row the next time, the row is not actually being added. Calling `updateView1()` fixes this by removing an empty/unfilled row in the UI.

https://user-images.githubusercontent.com/90279765/183509866-9d97557c-959f-4fda-886d-4fe50f1d816d.mov

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts (hunk 3 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: i meant these lines

```
this.partition.push(this.addPartitions());
this.condition.push(this.addConditions());

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts (hunk 2 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-participants/experiment-participants.component.ts:205]: 👍

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts (hunk 4 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts:522]: I'm not 100% sure if I did it correctly but removing the `this.queries.push(this.addMetric());` removed the default row in the Metrics table. The same changes for the other steps. I need confirmation from @ppratikcr7 if these changes are the correct way to remove the default rows.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts (hunk 3 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: As I commented below, I believe those lines are used to add default rows.

Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: @zackcl Correct

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts (hunk 4 above)
Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts:522]: @zackcl Correct

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts (hunk 3 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts:103]: @ppratikcr7 Thank you!

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts (hunk 4 above)
Comment (zackcl) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts:522]: @ppratikcr7 Thank you!

**Review by danoswaltCL — APPROVED** [review]

**Pull request merged**